Content.Client/StationRecords/GeneralStationRecordConsoleWindow.xaml.cs

```diff
@@ -33,20 +41,63 @@ public GeneralStationRecordConsoleWindow()
             if (!_isPopulating)
                 OnKeySelected?.Invoke(null);
         };
+
+        for (int i = 0; i < _filterTypes.Length; i++)
+        {
+            StationRecordsFilterType.AddItem(GetTypeFilterLocals(i));
+        }
+
+        StationRecordsFilterType.OnItemSelected += eventArgs =>
+        {
+            if (_currentFilterType != eventArgs.Id)
+            {
+                _currentFilterType = eventArgs.Id;
+                FilterListingOfRecords();
+            }
+        };
+
+        StationRecordsFilters.OnPressed += _ =>
+        {
+            FilterListingOfRecords(StationRecordsFiltersValue.Text);
+        };
+
+        StationRecordsFiltersReset.OnPressed += _ =>
+        {
+            StationRecordsFiltersValue.Text = "";
+            FilterListingOfRecords();
+        };
     }
 
     public void UpdateState(GeneralStationRecordConsoleState state)
     {
+        if (state.Filter != null) {
+            if (state.Filter.type != _filterTypes[_currentFilterType]) {
+                int findedIndex = Array.IndexOf(_filterTypes, state.Filter.type);
```

Comment:
you should just pass in the enum value into the method and then convert it from there instead of getting the index from the array and passing in the index

Reply:
and it is

Reply:
no i mean, passing the literal enum value instead of typecasting it and fetching the enum value from the stored array you have

Reply:
you can still typecast the enum value when selecting by ID though, since it's likely that the enum will still be in the same order that it was inserted in if doing `GetValues()` when populating the option button

Reply:
i made to sate type in var _currentFilterType